src/rviz/robot/robot_link.cpp

```diff
@@ -632,6 +632,13 @@ void RobotLink::createEntityForGeometryElement(const urdf::LinkConstSharedPtr& l
       {
         sub->setMaterial(default_material_);
       }
+      else
+      {
+          // create a new material copy for each instance of a RobotLink
+          Ogre::MaterialPtr mat = Ogre::MaterialPtr(new Ogre::Material(nullptr, material_name, 0, ROS_PACKAGE_NAME));
+          *mat = *sub->getMaterial();
```

Comment:
I have to admit I don't quite understand this line

Reply:
Create a new material instance and copy the existing material into this instance.

Reply:
ah, ok, thanks, I was reading to much into the first line/constructor